### DISCUSSION TITLE
Is LeetCode worth our time?

### DISCUSSION THREAD

**Discussion by Elchin-Novruzov** [community/community#110241] [Programming Help]
### Select Topic Area
#50380  #108135 #70105 #103002 #21512 

## Is LeetCode worth our time?

Which one is better? make a project which is helps you retain and learn new information or solve a lot of LeetCode questions? Just I want to know why a lot of algorithms is important to get a job in web development industry. I have 2 years of experience in web development and I used just few algorithms in my whole work life. I have been very huge projects (create a new SaaS tech, develop an e-govrement portal) and I used just some algorithms and data structures. Thats isn't mean I'm week at algorithms, I'm a Informathic and Math olympia champion and I can say if you want to be an web developer you don't need to became an algorithm expert just learn basic to medium level algorithms and keep your time for make new projects which one is helps you to learn new thinks or strengthens your knowledge.

**Comment by deonlobo** [comment]
LeetCode can be useful for improving algorithmic problem-solving skills, particularly for technical interviews that frequently cover algorithmic and data structure concepts. Its importance, however, is determined by your career objectives and the specific requirements of the positions you seek. While algorithmic proficiency is required for certain roles, such as software engineering at high-tech companies, extensive project work can also effectively demonstrate your skills. Balancing LeetCode practice with real-world projects allows you to apply your knowledge practically while also demonstrating your ability to solve larger, more complex problems. Finally, the emphasis on algorithms varies across web development roles, so tailoring your learning approach to your career goals and the demands of the positions you want can be a strategic decision.

**Comment by 4chref** [comment]
You bring up a valid point. While algorithms and data structures are important in computer science and can be beneficial in certain aspects of web development, they may not be the primary focus for everyone in the industry. The choice between focusing on projects that help you learn new concepts or solving a lot of LeetCode questions depends on your career goals, interests, and the specific requirements of the jobs you're targeting.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Elchin-Novruzov ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
